### PR TITLE
[KED-2006] Add e2e tests for backward compatibility check

### PR DESCRIPTION
**Pull request by 921kiyo**
Added an e2e test for the latest Kedro 0.16.* against the starter tag 0.16.5. This only ensures the backward compatibility for `kedro run` and not other CLI command. 

0.16.3 and 0.16.4 were not compatible because of the `jupyter_client` dependency (which we fixed in https://github.com/quantumblacklabs/kedro-starter-pandas-iris/pull/8/files)

### CODE REVIEW

File: features/steps/run_steps.py [hunk 1]
```diff
@@ -59,6 +61,21 @@ def run_kedro_pipeline(context):
     )
 
 
+@given('I have installed kedro version "{version}"')
```
Comment (921kiyo) [inline features/steps/run_steps.py:64]: Code is the same as https://github.com/quantumblacklabs/kedro-viz/blob/main/package/features/steps/cli_steps.py#L109

File: features/run.feature [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
   Scenario: Run a Kedro project created from kedro-starter-pandas-iris
+    Given I have installed kedro version "latest"
     Given I have prepared a config file
     And I have run a non-interactive kedro new with the starter
     And I have installed the Kedro project's dependencies
     And I have run the Kedro pipeline
     Then I should get a successful exit code
+
+  Scenario: Backward compatibility between latest Kedro 0.16.* and 0.16.5 starter
```
Comment (limdauto) [inline features/run.feature:11]: I would suggest factor out the version to be generic, i.e. get the current Kedro version and find the closest major version to check against.

Comment (DmitriiDeriabinQB) [reply features/run.feature:11]: Yes, I'm also not entirely sure why we want to test the _past_ version of the starter against the specific version of Kedro. 

Would it make more sense to just install Kedro latest, pull the starter without the `checkout` option and make sure that the run works?

**Comment by 921kiyo** [comment]
Not required anymore

**Pull request closed without merging**